### PR TITLE
feat(Matter): Adds New Matter Humidity Sensor Endpoint

### PR DESCRIPTION
**Pull request by SuGlider**
## Description of Change
Adds a new Matter Endpoint: Humidity Sensor.
This endpoint stores Relative Humidity (percent) information.

## Tests scenarios
ESP32-C6 and ESP32-S3 using the provided example

## Related links
Related to #7432

### CODE REVIEW

File: libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+#include <Matter.h>
+#include <WiFi.h>
+
+// List of Matter Endpoints for this Node
+// Matter Humidity Sensor Endpoint
+MatterHumiditySensor SimulatedHumiditySensor;
+
+// set your board USER BUTTON pin here - decommissioning button
+#ifdef BOOT_PIN
+const uint8_t buttonPin = BOOT_PIN;  // Set your pin here. Using BOOT Button.
```
Comment (me-no-dev) [inline libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino:32]: See my note on the other PR about BOOT_PIN always being defined

File: libraries/Matter/src/MatterEndpoints/MatterHumiditySensor.h [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    log_e("Humidity Sensor Percentage value out of range [0..100].");
+    return false;
+  }
+    return begin(static_cast<uint16_t>(humidityPercent * 100.0f));
+  }
+  // this will just stop processing Humidity Sensor Matter events
+  void end();
+
+  // set the humidity percent with 1/100th of a percent precision
+  bool setHumidityPercent(double humidityPercent) {
```
Comment (me-no-dev) [inline libraries/Matter/src/MatterEndpoints/MatterHumiditySensor.h:38]: similar to the temperature, humidity is "always" in percents, so leaving just `setHumidity` and `getHumidity` should be enough (and consistent with the temperature sensor).

File: libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino (hunk 1 above)
Comment (SuGlider) [reply libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino:31]: done.

File: libraries/Matter/src/MatterEndpoints/MatterHumiditySensor.h (hunk 2 above)
Comment (SuGlider) [reply libraries/Matter/src/MatterEndpoints/MatterHumiditySensor.h]: done

File: libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  }
+
+  if (digitalRead(buttonPin) == HIGH && button_state) {
+    button_state = false;  // released
+  }
+
+  // Onboard User Button is kept pressed for longer than 5 seconds in order to decommission matter node
+  uint32_t time_diff = millis() - button_time_stamp;
+  if (button_state && time_diff > decommissioningTimeout) {
+    // Factory reset is triggered if the button is pressed longer than 10 seconds
```
Comment (SuGlider) [inline libraries/Matter/examples/MatterHumiditySensor/MatterHumiditySensor.ino:125]:
```suggestion
```

**Review by lucasssvaz — APPROVED** [review]
LGTM

**Review by me-no-dev — APPROVED** [review]

**Pull request merged**